src/create/createWithOptions.test.ts

```diff
@@ -0,0 +1,247 @@
+	logo: { alt: "Test Alt", src: "test.png" },
+	mode: "create",
+	offline: false,
+	owner: "Test Owner",
+	preserveGeneratedFrom: false,
+	repository: "test-repo",
+	skipAllContributorsApi: false,
+	skipGitHubApi: false,
+	skipInstall: false,
+	skipRemoval: false,
+	skipRestore: false,
+	skipUninstall: false,
+	title: "Test Title",
+};
+
+const mock$ = vi.fn();
+
+vi.mock("execa", () => ({
+	get $() {
+		return mock$;
+	},
+}));
+
+const mockOctokit = new Octokit();
+const github = {
+	auth: "auth-token",
+	octokit: mockOctokit,
+};
+
+vi.mock("../shared/cli/spinners.js", () => ({
+	withSpinner: vi
+		.fn()
+		.mockImplementation(
+			async <Return>(label: string, task: SpinnerTask<Return>) => {
+				return await task();
+			},
+		),
+	withSpinners: vi
+		.fn()
+		.mockImplementation(
+			async (label: string, tasks: [string, SpinnerTask<unknown>][]) => {
+				for (const [, task] of tasks) {
+					await task();
+				}
+			},
+		),
+}));
+
+vi.mock("../steps/writing/writeStructure.js", () => ({
+	writeStructure: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../steps/writeReadme/index.js", () => ({
+	writeReadme: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../steps/finalizeDependencies.js", () => ({
+	finalizeDependencies: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../steps/runCommands.js", () => ({
+	runCommands: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../shared/doesRepositoryExist.js", () => ({
+	doesRepositoryExist: vi.fn().mockResolvedValue(true),
+}));
+
+vi.mock("../steps/initializeGitHubRepository/index.js", () => ({
+	initializeGitHubRepository: vi.fn().mockResolvedValue(true),
+}));
+
+vi.mock("../shared/getGitHubUserAsAllContributor.js", () => ({
+	getGitHubUserAsAllContributor: vi.fn().mockResolvedValue({
+		contributions: ["code", "doc"],
+		name: "Test User",
+	}),
+}));
+
+vi.mock("../steps/addToolAllContributors.js", () => ({
+	addToolAllContributors: vi.fn().mockResolvedValue(null),
+}));
+
+it("calls writeStructure with options", async () => {
+	await writeStructure(options);
+	expect(writeStructure).toHaveBeenCalledWith(options);
+});
+
+it("calls writeReadme with options", async () => {
+	await writeReadme(options);
+	expect(writeReadme).toHaveBeenCalledWith(options);
+});
+
+it("calls writeStructure and writeReadme with options", async () => {
+	await createWithOptions({ github, options });
+
+	expect(writeStructure).toHaveBeenCalledWith(options);
+	expect(writeReadme).toHaveBeenCalledWith(options);
+});
+
+it("calls addToolAllContributors with options", async () => {
+	await addToolAllContributors(options);
+	expect(addToolAllContributors).toHaveBeenCalledWith(options);
+});
+
+it("calls addToolAllContributors with options when excludeAllContributors is false", async () => {
+	options.excludeAllContributors = false;
+	options.skipAllContributorsApi = false;
+
+	await createWithOptions({ github, options });
+	expect(addToolAllContributors).toHaveBeenCalledWith(options);
```

Comment:
[Bug] Modifying objects shared between multiple tests is dangerous. It makes it possible that running tests in a different order or with `.skip`/`.only` result in different outcomes. It's safer to create new objects per-test that modify a shared base:

```suggestion
	const options = {
		...optionsBase,
		excludeAllContributors: false,
		skipAllContributorsApi: false,
	};

	await createWithOptions({ github, options });

	expect(addToolAllContributors).toHaveBeenCalledWith(options);
```

Here and elsewhere. Basically anything in a test that says something like `options.`.

Reply:
This makes sense, yes! I understand now. I will make this change!

Reply:
This makes sense, thanks for clarifying. I think I have identified all the changes to make!